### PR TITLE
fix: ReleaseBuffers issues

### PR DESCRIPTION
**Pull request by jachym-tousek-keboola**
**Changes:**
- Fix for #2058.

-----------

### CODE REVIEW

File: internal/pkg/service/stream/mapping/recordctx/fasthttp.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func (c *fastHTTPContext) HeadersMap() *orderedmap.OrderedMap {
 }
 
 func (c *fastHTTPContext) ReleaseBuffers() {
-	c.req.ResetBody()
+	c.req.Request.ResetBody()
```
Comment (jachym-tousek-keboola) [inline internal/pkg/service/stream/mapping/recordctx/fasthttp.go:86]: Turns out that `c.req.ResetBody()` actually calls `ctx.Response.ResetBody()`. We want to reset the request body here.

File: internal/pkg/service/stream/mapping/recordctx/http.go [hunk 2]
```diff
@@ -84,6 +84,7 @@ func (c *httpContext) HeadersMap() *orderedmap.OrderedMap {
 }
 
 func (c *httpContext) ReleaseBuffers() {
+	_ = c.req.Body.Close()
```
Comment (jachym-tousek-keboola) [inline internal/pkg/service/stream/mapping/recordctx/http.go:87]: This is more of a guess but shouldn't we close the request body `io.ReadCloser` here? We're not planning to read from it again. Fast http internally does this (and ignores the err).

File: internal/pkg/service/stream/mapping/recordctx/fasthttp.go (hunk 1 above)
Comment (Matovidlo) [reply internal/pkg/service/stream/mapping/recordctx/fasthttp.go:86]: https://github.com/keboola/keboola-as-code/pull/2058/files#r1778147284 I am not sure. Internally this calls the `c.req.Request.ResetBody()` but only when `keepBodyBuffer` is on. Are we sure that this eradicate the issue? I think with this approach we will perform another benchmark and conclude that it does not need to help

File: internal/pkg/service/stream/mapping/recordctx/http.go (hunk 2 above)
Comment (Matovidlo) [reply internal/pkg/service/stream/mapping/recordctx/http.go:87]: I don't think so if you want to reuse the buffers. It puts the body into the pool. If you do this you will reuse closed body and need to recreate the req.Body.
We can try it but I think this is not good approach. Could you test it locally with k6? You should have memory leak in the stream with previous version and some better performance .
https://github.com/keboola/keboola-as-code/blob/main/docs/stream/benchmarks.md

File: internal/pkg/service/stream/mapping/recordctx/fasthttp.go (hunk 1 above)
Comment (jachym-tousek-keboola) [reply internal/pkg/service/stream/mapping/recordctx/fasthttp.go:86]: > Internally this calls the `c.req.Request.ResetBody()`

It doesn't. You're in Response object here.

> Are we sure that this eradicate the issue? 

No. I don't think the issue is related to #2058 in the first place.

File: internal/pkg/service/stream/mapping/recordctx/http.go (hunk 2 above)
Comment (jachym-tousek-keboola) [reply internal/pkg/service/stream/mapping/recordctx/http.go:87]: I think the comment is unrelated to this line? `net/http` doesn't seem to be using any pool in the first place, only fasthttp does.

**Review by Matovidlo — APPROVED** [review]
LGTM

**Pull request merged**